api/types.go

```diff
@@ -157,6 +157,23 @@ func (tr Timerange) IsValid() bool {
 		tr.Start <= tr.End
 }
 
+// Round() will fix some invalid timeranges by rounding their starts and ends.
+func (tr Timerange) Round() Timerange {
+	if tr.Resolution == 0 {
+		return tr
+	}
+	tr.Start = (tr.Start + tr.Resolution/2) / tr.Resolution * tr.Resolution // This performs a round
+	tr.End = (tr.End + tr.Resolution/2) / tr.Resolution * tr.Resolution     // This performs a round
+	return tr
+}
+
+// Later() returns a timerange which is forward in time by the amount given
+func (tr Timerange) Later(time int64) Timerange {
```

Comment:
is `x.Start() - x.End()` guaranteed to have `y.Start() - y.End()`, where `y = x.Later(range)` for any range? I think the answer is yes, but I want to be confident that is the property we have (since time-shifted fetch should hopefully fetch the same amount of data)